snapcraft/_store.py

```diff
@@ -793,19 +793,78 @@ def validate(snap_name, validations, revoke=False, key=None):
         if revoke:
             assertion['revoked'] = "true"
 
-        assertion = _sign_validation(validation, assertion, key)
+        assertion = _sign_assertion(validation, assertion, key, 'validations')
 
         # Save assertion to a properly named file
         fname = '{}-{}-r{}.assertion'.format(snap_name, gated_name, rev)
         with open(fname, 'wb') as f:
             f.write(assertion)
 
-        store.push_validation(snap_id, assertion)
+        store.push_assertion(snap_id, assertion, endpoint='validations')
 
 
 validation_re = re.compile('^[^=]+=[0-9]+$')
 
 
+def collaborate(snap_name, key):
+    store = storeapi.StoreClient()
+
+    with _requires_login():
+        account_info = store.get_account_information()
+    publisher_id = account_info['account_id']
+
+    release = storeapi.constants.DEFAULT_SERIES
+    try:
+        snap_id = account_info['snaps'][release][snap_name]['snap-id']
+    except KeyError:
+        raise storeapi.errors.SnapNotFoundError(snap_name)
+    developers = _get_developers(snap_id, publisher_id)
+
+    # XXX: Do the amendments via UI here.
+    #
+    # The data will look like:
+    # {'snap_developer': {
+    #      'type: 'snap-developer',
+    #      'authority-id': <account_id of the publisher Or `canonical`>,
```

Comment:
Sort of. By the time the UI gets involved the `authority-id` must be changed to the publisher's id as that's whos key is signing the new assertion.